src/Illuminate/Cache/RedisStore.php

```diff
@@ -342,6 +355,10 @@ protected function serialize($value)
      */
     protected function unserialize($value)
     {
+        if ($this->strict && filter_var($value, FILTER_VALIDATE_INT)) {
```

Comment:
I think this needs to be:

```php
if ($this->strict && filter_var($value, FILTER_VALIDATE_INT) !== false) {
```

`filter_var('0', FILTER_VALIDATE_INT)` evaluates to `0` which is PHP falsy so the string `'0'` is unexpectedly returned.

Reply:
Thanks @derekmd Fixed